x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/pipeline_processors/create_ml_inference_pipeline.ts

```diff
@@ -56,9 +61,15 @@ export const createAndReferenceMlInferencePipeline = async (
     esClient
   );
 
+  const mappingResponse = fieldMappings
+    ? (await updateMlInferenceMappings(indexName, fieldMappings, esClient)).acknowledged
+    : false;
+
   return {
-    ...createPipelineResult,
-    addedToParentPipeline: addSubPipelineResult.addedToParentPipeline,
+    added_to_parent_pipeline: addSubPipelineResult.addedToParentPipeline,
+    created_pipeline: createPipelineResult.created,
+    mapping_updated: mappingResponse,
+    pipeline_id: createPipelineResult.id,
```

Comment:
Looks like we're changing the return type. Isn't this breaking [routes/enterprise_search/indices](https://github.com/elastic/kibana/pull/154291/files#diff-f9440a95c3204a819bc8f1fe51dafe281544bedc7083e0315795c5dc598cd933R463-R472)?

Reply:
Nope. The return of the API was returning the value `id` before and now is returning the value `pipeline_id`, which are the same value, just renamed property. And the returned response's key for this ID is not changing: https://github.com/elastic/kibana/pull/154291/files#diff-f9440a95c3204a819bc8f1fe51dafe281544bedc7083e0315795c5dc598cd933R475-R477